### PR TITLE
Fix #82

### LINKED ISSUE #82: SwinUNETR Pretraining ignores in_channels != 1 in reconstruction loss
**Describe the bug**
The SwinUNETR pretraining script accepts in_channels greater than 1, but the reconstruction loss calculation in this case is wrong since the model only outputs 1 channel, regardless of the given number.

**To Reproduce**
1. Construct a multichannel dataset compatible with SwinUNETR pretraining.
2. Run SwinUNTER pretraining with `--in_channels N` where N > 1.

**Expected behavior**
The output channels of the reconstruction head of the SwinUNETR should match the input channels.

**Screenshots**
image

**Environment (please complete the following information):**
 - OS: `amazon/Deep Learning AMI GPU PyTorch 1.12.0 (Ubuntu 20.04) 20220706`
 - Python version: `3.9.13`
 - MONAI version [e.g. git commit hash]: `0.9.1rc3+4.g5fb767c`
 - CUDA/cuDNN version: 11.6
 - GPU models and configuration: 1 x Tesla T4

### PR DESCRIPTION
**Pull request by ardapekis**
**Third-Party Contribution.**

Resolves Issue #82.

### CODE REVIEW

**Comment by wyli** [comment]
could you please help confirm @tangy5?

**Comment by tangy5** [comment]
Thanks, yes, that part is hard coded. The changes are good to add multi channels support.
